File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/ItemForm.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms

Partial Public Class ItemForm
	Inherits Form
	#Region "Public constructor"

	Public Sub New()
		InitializeComponent()
		OnIsReadOnlyChanged()
	End Sub

	#End Region

	#Region "Private methods"

	Private Sub OnIsReadOnlyChanged()
		btnOk.Visible = Not tbValue.ReadOnly
		btnCancel.Text = CStr(IIf((Not tbValue.ReadOnly), "Cancel", "Close"))
	End Sub

	#End Region

	#Region "Public properties"

	Public Property Value() As String
		Get
			Return tbValue.Text
		End Get
		Set(ByVal value As String)
			tbValue.Text = value
		End Set
	End Property

	Public Property IsReadOnly() As Boolean
		Get
			Return tbValue.ReadOnly
		End Get
		Set(ByVal value As Boolean)
			If tbValue.ReadOnly <> value Then
				tbValue.ReadOnly = value
				OnIsReadOnlyChanged()
			End If
		End Set
	End Property

	#End Region
End Class

File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/ItemForm.cs
using System.Windows.Forms;

namespace Neurotec.Samples
{
	public partial class ItemForm : Form
	{
		#region Public constructor

		public ItemForm()
		{
			InitializeComponent();
			OnIsReadOnlyChanged();
		}

		#endregion

		#region Private methods

		private void OnIsReadOnlyChanged()
		{
			btnOk.Visible = !tbValue.ReadOnly;
			btnCancel.Text = !tbValue.ReadOnly ? "Cancel" : "Close";
		}

		#endregion

		#region Public properties

		public string Value
		{
			get
			{
				return tbValue.Text;
			}
			set
			{
				tbValue.Text = value;
			}
		}

		public bool IsReadOnly
		{
			get
			{
				return tbValue.ReadOnly;
			}
			set
			{
				if (tbValue.ReadOnly != value)
				{
					tbValue.ReadOnly = value;
					OnIsReadOnlyChanged();
				}
			}
		}

		#endregion
	}
}
